server/src/main/java/org/opensearch/index/shard/RefreshListeners.java

```diff
@@ -256,51 +334,103 @@ public void afterRefresh(boolean didRefresh) throws IOException {
         /* Grab the current refresh listeners and replace them with null while synchronized. Any listeners that come in after this won't be
          * in the list we iterate over and very likely won't be candidates for refresh anyway because we've already moved the
          * lastRefreshedLocation. */
-        List<Tuple<Translog.Location, Consumer<Boolean>>> candidates;
```

Comment:
Theres a lot of duplication here we can refactor out, but at least I think we should split this into two sub functions, one for each type of listener?

Reply:
Sure I will try remove duplication here, but most of it is due to different listener types